core/iwasm/compilation/aot_emit_memory.c

```diff
@@ -202,9 +203,32 @@ aot_check_memory_overflow(AOTCompContext *comp_ctx, AOTFuncContext *func_ctx,
                     goto fail;
                 }
             }
+            if (alignp != NULL) {
```

Comment:
This code piece is only for the situation that `addr + offset` is pre-known and is inside the initial size of linear memory, but it is also suitable for `addr + offset` is pre-known and is inside the maximum size of linear memory, how about put these code before L184 `if (mem_offset + bytes <= mem_data_size)`?

Reply:
it makes sense.

Reply:
done